File: apps/api/schemas/chart.py
from pydantic import BaseModel


class ChartPointResponse(BaseModel):
    label: str
    value: float


class ChartSeriesResponse(BaseModel):
    id: str
    label: str
    color: str
    points: list[ChartPointResponse]


class ChartResponse(BaseModel):
    id: str
    title: str
    subtitle: str
    series: list[ChartSeriesResponse]

File: packages/types/chart.ts
export type ChartPoint = {
  label: string;
  value: number;
};

export type ChartSeries = {
  id: string;
  label: string;
  color: string;
  points: ChartPoint[];
};

export type CompanyChart = {
  id: string;
  title: string;
  subtitle: string;
  series: ChartSeries[];
};
